### PR TITLE
Implemented changes for new sorting policy in CLD

### PR DESCRIPTION
**Pull request by eleogran**
BEGINRELEASENOTES
- FCCee_o1_v04: Implemented changes the new sorting policy
   - Analogous of #234 

ENDRELEASENOTES

### CODE REVIEW

**Comment by eleogran** [comment]
@andresailer can you please cross-check?

File: FCCee/compact/FCCee_o1_v04/InnerTracker_o2_v06_02.xml [hunk 1]
```diff
@@ -490,37 +490,41 @@
...
+	  <layer id="7" inner_r="InnerTracker_Barrel_radius_2-2*cm+0.6/3.5959*cm" inner_z="InnerTracker_Barrel_half_length_2+2*cm" outer_r="InnerTracker_Barrel_radius-1*cm" vis="InterlinkVis">
+	    <slice material="CarbonFiber" thickness="0.6/3.5959*cm" />
+	  </layer>
+	  <layer id="8" inner_r="InnerTracker_Barrel_radius_0" inner_z="InnerTracker_Barrel_half_length_1+2*cm" outer_r="InnerTracker_Barrel_radius_1" vis="InterlinkVis">
+            <slice material="CarbonFiber" thickness="0.6/3.5959*cm" />
+          </layer>
+          <layer id="9" inner_r="InnerTracker_Barrel_radius_1" inner_z="InnerTracker_Barrel_half_length_1+2*cm" outer_r="InnerTracker_Barrel_radius_2-2*cm" vis="InterlinkVis">
+	    <slice material="CarbonFiber" thickness="0.6/3.5959*cm" />
+          </layer>
+        </layer>
```
Comment (andresailer) [inline FCCee/compact/FCCee_o1_v04/InnerTracker_o2_v06_02.xml]: this looks wrong

Comment (eleogran) [reply FCCee/compact/FCCee_o1_v04/InnerTracker_o2_v06_02.xml]: you mean layer id="7"?
It should be fixed now:
 <layer id="7" inner_r="InnerTracker_Barrel_radius_2-2*cm+0.6/3.5959*cm" inner_z="InnerTracker_Barrel_half_length_2+2*cm" outer_r="InnerTracker_outer_radius-1*cm" vis="InterlinkVis">

Comment (andresailer) [reply FCCee/compact/FCCee_o1_v04/InnerTracker_o2_v06_02.xml]: No I mean the two closing of layer: `</layer>`

Comment (eleogran) [reply FCCee/compact/FCCee_o1_v04/InnerTracker_o2_v06_02.xml]: fixed now

File: FCCee/compact/FCCee_o1_v04/OuterTracker_o2_v06_02.xml [hunk 2]
```diff
@@ -180,27 +180,30 @@
...
+	  <layer id="1" inner_r="OuterTracker_inner_radius" inner_z="OuterTracker_Endcap_z_1-1*cm" outer_r="OuterTracker_Endcap_outer_radius" vis="SupportVis">
+	    <slice material="CarbonFiber" thickness="0.365/3.5959*cm" />
+	  </layer>
+	  <layer id="2" inner_r="OuterTracker_inner_radius" inner_z="OuterTracker_Endcap_z_2+1*cm" outer_r="OuterTracker_Endcap_outer_radius" vis="SupportVis">
+	    <slice material="CarbonFiber" thickness="0.365/3.5959*cm" />
+	  </layer>
+	  <layer id="3" inner_r="OuterTracker_inner_radius" inner_z="OuterTracker_Endcap_z_3-1*cm" outer_r="OuterTracker_Endcap_outer_radius" vis="SupportVis">
+	    <slice material="CarbonFiber" thickness="0.365/3.5959*cm" />
+	  </layer>
+	  <layer id="4" inner_r="OuterTracker_Barrel_radius_0-1*cm" inner_z="OuterTracker_Barrel_half_length+2*cm" outer_r="OuterTracker_outer_radius_1-0.5*cm" vis="OuterTrackerInterlinkVis">
```
Comment (andresailer) [inline FCCee/compact/FCCee_o1_v04/OuterTracker_o2_v06_02.xml]:
```suggestion
	  <layer id="4" inner_r="OuterTracker_Barrel_radius_0-1*cm" inner_z="OuterTracker_Barrel_half_length+2*cm" outer_r="OuterTracker_outer_radius-0.5*cm" vis="OuterTrackerInterlinkVis">
```

Comment (eleogran) [reply FCCee/compact/FCCee_o1_v04/OuterTracker_o2_v06_02.xml]: done

**Comment by andresailer** [comment]
Please rebase, and maybe fix the commit message

**Comment by andresailer** [comment]
There are issues with the sorting policy arguments: the variable VertexEndcap_zmin is not defined

File: FCCee/compact/FCCee_o1_v04/FCCee_o1_v04.xml [hunk 3]
```diff
@@ -298,6 +298,59 @@
...
+
+        <argument value="/OuterTrackerEndcapSupport" />
+        <argument value="OuterTracker_Barrel_half_length" />
+        <argument value="OuterTracker_Barrel_radius_1+3*cm" />
+        <argument value="(OuterTracker_Endcap_radius_2-OuterTracker_Barrel_radius_1)/(OuterTracker_half_length-OuterTracker_Barrel_half_length)" />
+
+        <argument value="/VertexEndcap/" />
+        <argument value="VertexEndcap_zmin" />
+        <argument value="VertexBarrel_r3+1*cm" />
+        <argument value="(VertexEndcap_rmax-VertexBarrel_r3+1*cm)/(VertexEndcap_zmax-VertexEndcap_zmin)" />
```
Comment (andresailer) [inline FCCee/compact/FCCee_o1_v04/FCCee_o1_v04.xml]: VertexEmdcap_zmin and VertexEmdcap_zmax are not defined

Comment (eleogran) [reply FCCee/compact/FCCee_o1_v04/FCCee_o1_v04.xml]: replaced with VertexEndcap_z1 and VertexEndcap_z3 respectively

File: FCCee/compact/FCCee_o1_v04/OuterTracker_o2_v06_02.xml (hunk 2 above)
Comment (andresailer) [reply FCCee/compact/FCCee_o1_v04/OuterTracker_o2_v06_02.xml]: This was actually correct before my suggestion (#280)

**Pull request merged**